From e7a73040b4faf447fec4806a2670d68127b73cfa Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@altera.com>
Date: Thu, 6 Mar 2014 16:22:27 -0600
Subject: [PATCH 026/142] FogBugz #99945-2: add gpio-dwapb driver to defconfig
 and dts

* Add the gpio-dwapb driver to the defconfig and device tree.
* Fix LED definitions in device tree
* Move the LED definitions that are specific to the Altera
  CycloneV socdk board to the proper device tree file.

Signed-off-by: Alan Tull <atull@altera.com>

v4: remove socfpga.dtsi changes which were upstreamed
    fix led definitions to match upstream socfpga.dtsi
    enable gpios in cyclone5 and arria5 dtsi
v3: fix arria5 socdk led definitions
    move cyclone5 led definitions to cyclone5 socdk dts
    move clock property to gpio main node, not gpio port nodes
v2: fix cyclone5 led definitions

Conflicts:

	arch/arm/boot/dts/socfpga_arria5.dtsi
	arch/arm/boot/dts/socfpga_cyclone5.dtsi
[Original patch taken from
 https://github.com/altera-opensource/linux-socfpga.git
 tag rel_socfpga-4.8_17.03.01_pr]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/arm/boot/dts/socfpga_arria5_socdk.dts   | 17 +++++++++++++
 arch/arm/boot/dts/socfpga_cyclone5_socdk.dts | 36 ++++++++++++++++++++++++++++
 arch/arm/configs/socfpga_defconfig           |  1 +
 3 files changed, 54 insertions(+)

diff --git a/arch/arm/boot/dts/socfpga_arria5_socdk.dts b/arch/arm/boot/dts/socfpga_arria5_socdk.dts
index 3c8867862b0d..58c493aa653e 100644
--- a/arch/arm/boot/dts/socfpga_arria5_socdk.dts
+++ b/arch/arm/boot/dts/socfpga_arria5_socdk.dts
@@ -61,8 +61,21 @@
 	rxc-skew-ps = <2000>;
 };
 
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
+	speed-mode = <0>;
 
 	eeprom@51 {
 		compatible = "atmel,24c32";
@@ -82,6 +95,10 @@
 	status = "okay";
 };
 
+&uart0 {
+	status = "okay";
+};
+
 &usb1 {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
index 15e43f43f244..873c82b49554 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
+++ b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
@@ -45,6 +45,29 @@
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 	};
+
+	leds {
+		compatible = "gpio-leds";
+		hps0 {
+			label = "hps_led0";
+			gpios = <&portb 15 1>;
+		};
+
+		hps1 {
+			label = "hps_led1";
+			gpios = <&portb 14 1>;
+		};
+
+		hps2 {
+			label = "hps_led2";
+			gpios = <&portb 13 1>;
+		};
+
+		hps3 {
+			label = "hps_led3";
+			gpios = <&portb 12 1>;
+		};
+	};
 };
 
 &gmac1 {
@@ -61,12 +84,21 @@
 	rxc-skew-ps = <2000>;
 };
 
+&gpio0 {
+	status = "okay";
+};
+
 &gpio1 {
 	status = "okay";
 };
 
+&gpio2 {
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
+	speed-mode = <0>;
 
 	eeprom@51 {
 		compatible = "atmel,24c32";
@@ -87,6 +119,10 @@
 	status = "okay";
 };
 
+&uart0 {
+	status = "okay";
+};
+
 &usb1 {
 	status = "okay";
 };
diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index 7816657aee59..124f5fe36178 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -92,6 +92,7 @@ CONFIG_I2C_DESIGNWARE_PLATFORM=y
 CONFIG_GPIOLIB=y
 CONFIG_GPIO_SYSFS=y
 CONFIG_GPIO_DWAPB=y
+CONFIG_GPIO_ALTERA=m
 CONFIG_PMBUS=y
 CONFIG_SENSORS_LTC2978=y
 CONFIG_SENSORS_LTC2978_REGULATOR=y
-- 
2.11.0

